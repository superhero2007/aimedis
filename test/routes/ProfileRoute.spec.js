import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import ProfileRoute from 'routes/ProfileRoute';

describe('routes/ProfileRoute', () => {
  it('should render Profile Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/profile']}>
        <ProfileRoute
          exact
          path="/profile"
          component={() => (<div>PROFILE</div>)}
          isAuthenticated={false}
        />
      </Router>);

    expect(render).toMatchSnapshot();
  });

  it('should render Profile Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/profile']}>
        <ProfileRoute
          exact
          path="/profile"
          component={() => (<div>PROFILE</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
