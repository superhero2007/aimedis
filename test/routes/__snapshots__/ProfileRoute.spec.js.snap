// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/ProfileRoute should render Profile Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">PROFILE</div>"`;

exports[`routes/ProfileRoute should render Profile Page for unauthenticated access 1`] = `"<div id=\\"redirect\\" data-reactroot=\\"\\"></div>"`;
